modules/virtualization/microvm/guivm.nix

```diff
@@ -0,0 +1,128 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  configHost = config;
+  waypipe-ssh = pkgs.callPackage ../../../user-apps/waypipe-ssh {};
+  guivmBaseConfiguration = {
+    imports = [
+      ({
+        lib,
+        pkgs,
+        ...
+      }: {
+        ghaf = {
+          users.accounts.enable = lib.mkDefault configHost.ghaf.users.accounts.enable;
+          profiles.graphics.enable = true;
+          profiles.applications.enable = false;
+          windows-launcher.enable = false;
+          development = {
+            # NOTE: SSH port also becomes accessible on the network interface
+            #       that has been passed through to NetVM
+            ssh.daemon.enable = lib.mkDefault configHost.ghaf.development.ssh.daemon.enable;
+            debug.tools.enable = lib.mkDefault configHost.ghaf.development.debug.tools.enable;
+          };
+        };
+
+        environment = {
+          etc = {
+            "ssh/waypipe-ssh".source = "${waypipe-ssh}/keys/waypipe-ssh";
+          };
+          systemPackages = [
+            pkgs.waypipe
+          ];
+        };
+
+        networking.hostName = "guivm";
+        system.stateVersion = lib.trivial.release;
+
+        nixpkgs.buildPlatform.system = configHost.nixpkgs.buildPlatform.system;
+        nixpkgs.hostPlatform.system = configHost.nixpkgs.hostPlatform.system;
+
+        networking = {
+          enableIPv6 = false;
+          interfaces.ethint0.useDHCP = false;
+          firewall.allowedTCPPorts = [22];
+          firewall.allowedUDPPorts = [67];
+          useNetworkd = true;
+        };
+
+        microvm = {
+          mem = 2048;
+          hypervisor = "qemu";
+          qemu.bios.enable = false;
+          storeDiskType = "squashfs";
+          interfaces = [
+            {
+              type = "tap";
+              id = "vm-guivm";
+              mac = "02:00:00:02:02:02";
+            }
+          ];
+        };
+
+        networking.nat = {
+          enable = true;
+          internalInterfaces = ["ethint0"];
+        };
+
+        # Set internal network's interface name to ethint0
+        systemd.network.links."10-ethint0" = {
+          matchConfig.PermanentMACAddress = "02:00:00:02:02:02";
+          linkConfig.Name = "ethint0";
+        };
+
+        systemd.network = {
+          enable = true;
+          networks."10-ethint0" = {
+            matchConfig.MACAddress = "02:00:00:02:02:02";
+            addresses = [
+              {
+                # IP-address for debugging subnet
+                addressConfig.Address = "192.168.101.3/24";
+              }
+            ];
+            routes = [
+              {routeConfig.Gateway = "192.168.101.1";}
```

Comment:
Earlier comment on subnet applies here - DRY with `Gateway` and when we want to change the VM addresses, one place for network declarations could help. For new, out of ghaf (ghaf as lib), VM declarations, users could still use the platform subnet + gateway but still set their VM static address in their own VMs.

Reply:
SP-2839 is created to track this